### PR TITLE
Add code coverage

### PR DESCRIPTION
**Pull request by eessex**
Addresses https://artsyproduct.atlassian.net/browse/PLATFORM-1137

This PR adds code coverage reports to both Mocha and Jest test suites. Coverage appears in the console when running tests locally and in CI. Reports are also uploaded to https://codecov.io/gh/artsy 

Coverage reports for this PR can be viewed here: https://codecov.io/gh/artsy/force/pull/3930/tree

The `codecov` library is used only to upload reports, and is run with the script `report-coverage`. This step is invoked during CI runs and creates a few new env vars for the test run. 

To generate reports in Mocha, the `nyc` library is used in combination with `mocha-lcov-reporter`. Jest uses its built in coverage. Controls for both are located in `package.json`. We can add more options there, including minimum coverage for the repo, or other flags that will cause an ❌in CI.

### CODE REVIEW

**Comment by eessex** [comment]
Coverage reports for this PR can be viewed here: https://codecov.io/gh/artsy/force/pull/3930/tree

File: package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "sync-schema": "rm -rf data && mkdir data && graphql-fetch-schema https://metaphysics-staging.artsy.net -o data",
     "test": "scripts/test.sh",
     "type-check": "tsc --pretty --noEmit",
-    "webpack": "node --max_old_space_size=4096 -r dotenv/config -r @babel/register node_modules/.bin/webpack --config ./webpack"
+    "webpack": "node --max_old_space_size=4096 -r dotenv/config -r @babel/register node_modules/.bin/webpack --config ./webpack",
+    "report-coverage": "./node_modules/.bin/codecov -c $COMMIT_HASH_FULL -b $BRANCH_NAME"
```
Comment (damassi) [inline package.json]: alpha pls

File: .circleci/config.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ jobs:
       - yarn/setup
       - run:
           name: Test
-          command: yarn test
+          command: COMMIT_HASH_FULL=$(git rev-parse HEAD) CODECOV_TOKEN=$CODECOV_TOKEN BRANCH_NAME=$CIRCLE_BRANCH yarn test
```
Comment (damassi) [inline .circleci/config.yml]: Out of curiosity why do we need the `COMMIT_HASH` again? We were able to remove this when we migrated off of bucket-assets.

Comment (eessex) [reply .circleci/config.yml]: We send this to codecov to upload reports, this is the format of what we send them: 
`
{ commit: 'd41b687186251342927045b4fb91ca238d139c4c',
  branch: 'pull/3930',
  package: 'node-v3.3.0' }
`

It is also used in the urls for reports like this one: https://codecov.io/github/artsy/force/commit/d41b687186251342927045b4fb91ca238d139c4c

The var only lives in the circle test step, and is added to the `.env.test` but we do not ever need to set it anywhere.

Comment (damassi) [reply .circleci/config.yml]: Sounds good - ty for explaining 👍

**Pull request merged**